### PR TITLE
New comparators in OPL

### PR DESCRIPTION
**Pull request by riya-17**
AAP-15531

* Allow less than max
* Allow greater than min

### CODE REVIEW

File: opl/investigator/check.py [hunk 1]
```diff
@@ -42,6 +42,52 @@ def _check_by_min_max(data, value):
     return lower_boundary <= value <= upper_boundary, info
 
 
+def _check_by_max(data, value):
```
Comment (kdelee) [inline opl/investigator/check.py]: suggest name `_check_by_lte_max`

File: opl/investigator/check.py [hunk 2]
```diff
@@ -42,6 +42,52 @@ def _check_by_min_max(data, value):
...
+    logging.debug(f"data={data} and value={value}")
+    mean = statistics.mean(data)
+    lower_boundary = float(mean - (mean - min(data)))
```
Comment (kdelee) [inline opl/investigator/check.py:33]: would `lower_boundary = None` work? or does it freak out
could also put `lower_boundary` to -1 or 0 to indicate there is none

File: opl/investigator/check.py (hunk 1 above)
Comment (riya-17) [reply opl/investigator/check.py:45]: Same for min, `_check_by_lte_min`?

File: opl/investigator/check.py [hunk 3]
```diff
@@ -42,6 +42,52 @@ def _check_by_min_max(data, value):
...
+            ("data min", float(min(data))),
+            ("data max", float(max(data))),
+            ("lower_boundary", lower_boundary),
+            ("upper_boundary", upper_boundary),
+        ]
+    )
+    return value <= upper_boundary, info
+
+
+def _check_by_min(data, value):
```
Comment (kdelee) [inline opl/investigator/check.py]: suggest name `_check_by_gte_min`

File: opl/investigator/check.py [hunk 4]
```diff
@@ -42,6 +42,52 @@ def _check_by_min_max(data, value):
...
+    logging.debug(f"data={data} and value={value}")
+    mean = statistics.mean(data)
+    lower_boundary = float(mean - (mean - min(data)))
+    upper_boundary = float(mean + (max(data) - mean))
```
Comment (kdelee) [inline opl/investigator/check.py]: same idea, maybe `upper_boundary = None` or set it to `-1` or `99999999999` something big to communicate no upper boundary depending on what framework accepts w/o freaking out

File: opl/investigator/check.py (hunk 2 above)
Comment (riya-17) [reply opl/investigator/check.py:33]: I believe None would work. I thought that we are not comparing but we can still report lower boundary.

File: opl/investigator/check.py [hunk 5]
```diff
@@ -98,6 +144,13 @@ def check_by_min_max_0_1(data, value):
     """Checks if the current value is within the min/max range of previous values"""
     return _check_by_min_max(data, value)
 
+def check_by_max(data, value):
```
Comment (kdelee) [inline opl/investigator/check.py]: `check_by_lte_max`

File: opl/investigator/check.py [hunk 6]
```diff
@@ -98,6 +144,13 @@ def check_by_min_max_0_1(data, value):
...
+    """Checks if the current value is less than max range of previous values"""
+    return _check_by_max(data, value)
+
+def check_by_min(data, value):
```
Comment (kdelee) [inline opl/investigator/check.py]: `check_by_gte_min`

**Review by kdelee — APPROVED** [review]
LGTM!

**Pull request merged**